### PR TITLE
Spellcheck spans

### PR DESCRIPTION
**Pull request by sbaughman**

### CODE REVIEW

File: plugins/nanospell/plugin.js [hunk 1]
```diff
@@ -887,12 +885,6 @@
 			range.extractContents().appendTo(span);
 			range.insertNode(span);
 		},
-		markTypos: function (editor, node) {
-			var range = editor.createRange();
-			range.selectNodeContents(node);
-
-			this.markTyposInRange(editor, range);
-		},
```
Comment (caffodian) [inline plugins/nanospell/plugin.js:895]: I'm surprised that this could go away.  Then I noticed that in scanWords, we duplicate this functionality.

Comment (caffodian) [reply plugins/nanospell/plugin.js:895]: I think I'd rather keep this method and then have scanWords call it instead.  Mostly for clarity reasons, and also, I suspect that in the future, we may have some cases where we'll want to be smarter about doing element checking vs. actual range checking.  (For example, copy/paste or drag and drop, which don't happen as often, but we haven't really thought about at all)

File: plugins/nanospell/plugin.js [hunk 2]
```diff
@@ -919,17 +911,6 @@
...
-			var range = editor.createRange(),
-				block;
-
-			range.selectNodeContents(editor.editable());
-
-			var iterator = range.createIterator();
-			while (( block = iterator.getNextParagraph() )) {
-				this.markTypos(editor, block);
-			}
-		},
```
Comment (caffodian) [inline plugins/nanospell/plugin.js:932]: Ah yes.  I noticed this function was dead code as well, after the previous thing.

**Review by caffodian — COMMENTED** [review]
looks good to me, pending test revisions

File: plugins/nanospell/plugin.js [hunk 3]
```diff
@@ -267,6 +254,9 @@
 				}
 				return true;
 			});
+			editor.on(EVENT_NAMES.START_RENDER, render, self);
+			editor.on(EVENT_NAMES.START_SCAN_WORDS, scanWords, self);
+			editor.on(EVENT_NAMES.START_CHECK_WORDS, checkWords, self);
```
Comment (caffodian) [inline plugins/nanospell/plugin.js:262]: curious why you moved these.  Not opposed.  I was floating between these two options for a while.

Comment (sbaughman) [reply plugins/nanospell/plugin.js:262]: Just seemed to make sense to have all the event listener setup together.  I also like the pattern of making a `setupEventListeners` function to group them together and calling that from `init`

File: plugins/nanospell/plugin.js [hunk 4]
```diff
@@ -96,12 +96,16 @@
...
+			// or another element that logically should cause a word break
+
 			if (!condition) {
-				if (isNotBookmark(node) &&
+				if (node.type == CKEDITOR.NODE_ELEMENT &&
+					isNotBookmark(node) &&
 				 	((blockLimit && !blockLimitIsStartNode) ||
-					(block && !blockIsStartNode))) {
+					(block && !blockIsStartNode) ||
+					node.is('br') || node.is('sup') || node.is('sub'))) {
```
Comment (caffodian) [inline plugins/nanospell/plugin.js]: we can actually do

`node.is('br', 'sup', 'sub')`

at least this is what http://docs.ckeditor.com/#!/api/CKEDITOR.dom.element-method-is would indicate.

**Review by caffodian — CHANGES_REQUESTED** [review]
Could we also add a simple walking test, maybe something like

`<p>sometext<br/>something<sup>superscript</sup> anotherthing<sub>subscript</sub>` ?

The change looks simple, but it would be good for future regressions.  Looks good though!

**Review by caffodian — APPROVED** [review]

**Pull request merged**